### PR TITLE
Handle grayscale images and Theano vs. TF channel location difference

### PR DESCRIPTION
**Pull request by dan-1d**
Two features have been added:
1) images can be grayscale. Previously the input channels was assumed to be 3, which causes models to fail that expect 1 input channel.
2) Tensorflow and Theano use different ordering for the channel dimension. This fix corrects for it.

### CODE REVIEW

File: quiver_engine/server.py [hunk 1]
```diff
@@ -24,7 +24,16 @@
...
     get_evaluation_context = get_evaluation_context_getter()
-    single_input_shape = model.get_input_shape_at(0)[1:3]
+
+    if keras.backend.backend() == 'tensorflow':
+        single_input_shape = model.get_input_shape_at(0)[1:3]
+        input_channels = model.get_input_shape_at(0)[3]
+    elif keras.backend.backend() == 'theano':
+        single_input_shape = model.get_input_shape_at(0)[2:4]
+        input_channels = model.get_input_shape_at(0)[1]
+    else:
```
Comment (jakebian) [inline quiver_engine/server.py]: This else is unnecessary - afaik .backend() will never return anything else, I'd just change the elif to an else

File: quiver_engine/server.py [hunk 2]
```diff
@@ -50,11 +59,10 @@ def get_board_files(path):
     @app.route('/inputs')
     def get_inputs():
         image_regex = re.compile(r".*\.(jpg|png|gif)$")
+        files = [filename for filename in listdir(input_folder) if image_regex.match(filename) != None ]
+        return jsonify(files)
```
Comment (jakebian) [inline quiver_engine/server.py]: This is just reformatting?  I'd leave these couple of lines. I'm a sucker for function composition over declaring variables that only get over used once. More functional, less potentially mutable state.

File: quiver_engine/server.py [hunk 3]
```diff
@@ -70,17 +78,20 @@ def get_config():
 
     @app.route('/layer/<layer_name>/<input_path>')
     def get_layer_outputs(layer_name, input_path):
+        is_grayscale = (True if input_channels == 1 else False)
```
Comment (jakebian) [inline quiver_engine/server.py]: or simply ```is_grayscale = (input_channels == 1)```

File: quiver_engine/server.py [hunk 4]
```diff
@@ -95,7 +106,8 @@ def get_layer_outputs(layer_name, input_path):
         return jsonify(output_files)
     @app.route('/predict/<input_path>')
     def get_prediction(input_path):
-        input_img = load_img(input_path, single_input_shape)
+        is_grayscale = (True if input_channels == 1 else False)
```
Comment (jakebian) [inline quiver_engine/server.py]: same here

File: npm-debug.log [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+8 error node v5.2.0
+9 error npm  v3.3.12
+10 error path /Users/gebhardt/git/quiver/package.json
+11 error code ENOENT
+12 error errno -2
+13 error syscall open
+14 error enoent ENOENT: no such file or directory, open '/Users/gebhardt/git/quiver/package.json'
+14 error enoent This is most likely not a problem with npm itself
+14 error enoent and is related to npm not being able to find a file.
+15 verbose exit [ -2, true ]
```
Comment (jakebian) [inline npm-debug.log]: Remove the file please!

**Comment by dan-1d** [comment]
sorry.. was working on removing it as I got this

**Comment by jakebian** [comment]
Great, can you squash the last 3 commits though? The last 2 commits are not very meaningful.

File: quiver_engine/server.py (hunk 2 above)
Comment (dan-1d) [inline quiver_engine/server.py]: Reverted this. I was debugging another issue with the image file path.

**Comment by jakebian** [comment]
Awesome - I'll test it out before merging. Thanks @dan-1d!

**Pull request merged**